### PR TITLE
Rename createOne to new

### PR DESCRIPTION
**Pull request by daurnimator**
For #1872

### CODE REVIEW

**Comment by andrewrk** [comment]
This should probably also rename `destroy` to be `delete`.

File: test/tests.zig [hunk 1]
```diff
@@ -119,39 +115,36 @@ pub fn addCliTests(b: *build.Builder, test_filter: ?[]const u8, modes: []const M
...
 pub fn addTranslateCTests(b: *build.Builder, test_filter: ?[]const u8) *build.Step {
-    const cases = b.allocator.create(TranslateCContext{
-        .b = b,
-        .step = b.step("test-translate-c", "Run the C transation tests"),
-        .test_index = 0,
-        .test_filter = test_filter,
-    }) catch unreachable;
+    const cases = b.allocator.new(TranslateCContext) catch unreachable;
+    cases.b = b;
+    cases.step = b.step("test-translate-c", "Run the C transation tests");
```
Comment (mgxm) [inline test/tests.zig]: transation or translation?

**Comment by daurnimator** [comment]
> This should probably also rename `destroy` to be `delete`.

I don't know about that; I don't see it as necessary for this PR.
Also in another open PR I have a change in that area: https://github.com/ziglang/zig/pull/1863/commits/b3528a62f0cc85baf355ddec12b8d2f47a6475df

**Comment by andrewrk** [comment]
> I don't know about that; I don't see it as necessary for this PR.

We're talking about the API for allocation, right?

Status quo: `create` / `destroy`
After this change: `new` / `destroy`

Isn't `delete` the word that goes with `new`?

**Comment by daurnimator** [comment]
> Status quo: create / destroy

Really it's `createOne` / `destroy`. My desire was only to make it a single word, I'd be fine renaming it to `create` but that would cause upgrade issues.

We also have `free` which is *almost* the same as destroy.

**Comment by mgxm** [comment]
I tried to reproduce the failed build on FreeBSD without success I don't think it's related to this PR.
https://builds.sr.ht/~mgxm/job/20414

We could resubmit the build again and see if it works.

**Comment by daurnimator** [comment]
btw, this latest revision is no longer a breaking change; so please remove the label @andrewrk

**Comment by andrewrk** [comment]
I think it's too early for this change. Let's revisit this along with other naming changes closer to 1.0.0.

I definitely would want to either keep it as `create`/`destroy` or `new`/`delete` but not something in between.

**Pull request closed without merging**